File: impl/src/org/rr0/im/business/event/BirthImpl.java
package org.rr0.im.business.event;

import org.rr0.im.business.actor.Being;
import org.rr0.im.business.actor.HumanBeingImpl;
import org.rr0.im.business.event.circumstance.Moment;
import org.rr0.im.business.event.circumstance.Duration;
import org.rr0.im.business.event.Birth;
import org.rr0.im.service.function.classification.Category;
import org.rr0.im.service.function.classification.Classification;

/**
 * Being's birth Reference Implementation.
 *
 * @author Jérôme Beau
 * @version 21 avr. 2003 15:40:27
 */
public class BirthImpl extends EventImpl implements Birth
{
    private Being being;
    public BirthImpl(Being somePeople, Moment someDate) {
        super("Birth of " + somePeople, someDate);
    }

    public Category getForcedCategory (Classification classification) {
        return null;
    }

    public Duration getDuration () {
        return null;  // TODO
    }

    public Moment getDate() {
        return getEnd();
    }
}

File: impl/src/org/rr0/im/business/event/BirthImpl.ts
import {Moment} from "../../../../../../../api/src/org/rr0/im/business/event/circumstance/Moment";
import {Duration} from "../../../../../../../api/src/org/rr0/im/business/event/circumstance/Duration";
import {Category} from "../../../../../../../api/src/org/rr0/im/service/function/classification/Category";
import {Classification} from "../../../../../../../api/src/org/rr0/im/service/function/classification/Classification";
import {Being} from "../actor/BeingImpl";
import {Event} from "./Event";

/**
 * Being's birth Reference Implementation.
 */
export class BirthImpl extends Event {
  private being: Being;

  constructor(somePeople: Being, someDate: Moment) {
    super("Birth of " + somePeople, someDate);
  }

  public getForcedCategory(classification: Classification): Category {
    return null;
  }

  public getDuration(): Duration {
    return null;  // TODO
  }

  public getDate(): Moment {
    return getEnd();
  }
}
